crates/core/tedge/src/cli/connect/c8y_direct_connection.rs

```diff
@@ -111,22 +112,34 @@ fn publish_device_create_message(
 
 fn load_root_certs(
     root_store: &mut rustls_0_19::RootCertStore,
-    cert_dir: FilePath,
+    cert_path: FilePath,
 ) -> Result<(), ConnectError> {
-    for file_entry in fs::read_dir(cert_dir)? {
-        let file = file_entry?;
-        let f = File::open(file.path())?;
-        let mut rd = BufReader::new(f);
-        let _ = root_store.add_pem_file(&mut rd).map(|_| ()).map_err(|()| {
-            Error::new(
-                ErrorKind::InvalidData,
-                "could not load PEM file".to_string(),
-            )
-        });
+    if fs::metadata(&cert_path)?.is_dir() {
+        for file_entry in fs::read_dir(cert_path)? {
+            let file = file_entry?;
+            add_root_cert(root_store, file.path())?;
+        }
```

Comment:
Minor:
```suggestion
        for file_entry in fs::read_dir(cert_path)?? {
            add_root_cert(root_store, file_entry.path())?;
        }
```

Reply:
This has to be `file_entry?.path()`. Because `file_entry` is a `result`.